pre2p/network.go

```diff
@@ -0,0 +1,89 @@
+package pre2p
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"pocket/pre2p/types"
+)
+
+var _ types.Network = &network{}
+
+type network struct {
+	AddrBook []*types.NetworkPeer
+}
+
+func ConnectToValidatorNetwork(validators types.ValMap) (n types.Network) {
+	n = &network{}
+	for nodeId, v := range validators {
+		err := n.(*network).connectToValidator(nodeId, v)
+		if err != nil {
+			log.Println("[WARN] Error connecting to validator: ", err)
+			continue
+		}
+	}
+	return
+}
+
+func (n *network) NetworkBroadcast(data []byte, self types.NodeId) error {
+	// TODO(team): This address book is currently static and does not update dynamically as new peers come on/offline.
+	for _, peer := range n.AddrBook {
+		// TODO(team): Discuss how self-broadcasts should be handled. A the moment, the consensus
+		// module has custom logic that makes a leader take extra actions to also behave as a replica
+		// rather than having it be "generalized" through the P2P layer.
+		if self == peer.NodeId {
+			continue
+		}
+		client, err := net.DialTCP("tcp", nil, peer.ConsensusAddr)
+		if err != nil {
+			log.Println("Error connecting to peer: ", err)
+			continue
+		}
+		client.Write(data)
+		client.Close()
+	}
+	return nil
+}
+
+func (n *network) NetworkSend(data []byte, node types.NodeId) error {
+	for _, peer := range n.AddrBook {
+		// TODO(olshansky): Quick hack to avoid sending network messages to self.
+		if node != peer.NodeId {
+			continue
+		}
+		client, err := net.DialTCP("tcp", nil, peer.ConsensusAddr)
+		if err != nil {
+			log.Println("Error connecting to peer: ", err)
+			continue
+		}
+		client.Write(data)
```

Comment:
Need error handling. Don't forget to close the client

Reply:
Done.